### ISSUE TITLE
Mock Database fails on filtering the Demo User

### ISSUE THREAD

**Issue by liwde** [NGO-DB/ndb-core#167]
mock-database.ts errors in line 147, because the filter function calls .getType() on all objects in the database. This function does not exist on the Demo User as it is just a plain object and not set through the Entity Manager. (mock-database-manager.service.ts line 47)

This can be easily changed.

**Comment by sleidig** [comment]
PR #168 looks like a good, general improvement. When did this error occur? I couldn't find the mentioned `getType()` call on line 147.

**Comment by liwde** [comment]
line 147 executes the filter function constructed a few lines above on all objects in the db, so for example:

```ts
case 'attendances_index/by_month':
    filter = (e: AttendanceMonth) => e.getType() === AttendanceMonth.ENTITY_TYPE && e.month === options.key;
    break;
```

**Comment by sleidig** [comment]
Thanks!

I think this is fixed with PR #168
